pkg/ipnet/ipnet.go

```diff
@@ -48,11 +48,11 @@ func (ipnet *IPNet) UnmarshalJSON(b []byte) (err error) {
 		return err
 	}
 
-	ip, net, err := net.ParseCIDR(cidr)
+	_, net, err := net.ParseCIDR(cidr)
 	if err != nil {
 		return err
 	}
-	ipnet.IP = ip
-	ipnet.Mask = net.Mask
+	ipnet.IPNet = *net
```

Comment:
This is lossy, and doesn't allow for an IP within the subnet.  I think we want to keep this package how it was.

Reply:
I think that's okay. IPNet is supposed to encode a network address, not a host address. Do we have any configuration options that allow a host address?

Reply:
> IPNet is supposed to encode a network address, not a host address. Do we have any configuration options that allow a host address?

Maybe not yet, but [recovering the initial IP seems straightforward][1], so I don't think we need to make project-specific assumptions in this package.

[1]: https://github.com/openshift/installer/pull/310#discussion_r219725415

Reply:
We should change the name of the package then, since the IPNet from the standard library is designed for the same task (the only functionality we are adding is the ability to marshal and unmarshal). I think that's probably out of scope for this PR, but I can follow up after this merges.

Reply:
> We should change the name of the package then, since the IPNet from the standard library is designed for the same task (the only functionality we are adding is the ability to marshal and unmarshal).

We hust add the pretty CIDR serialization.  The stdlib's `IPNet` [round-trips non-root IPs fine][1].

[1]: https://play.golang.org/p/1Cf3-GybaKO

Reply:
@wking the round trip from `pkg/ipnet`
https://play.golang.org/p/4-DXPIkNHsi

```console
2009/11/10 23:00:00 original: net.IPNet{IP:net.IP{0xc0, 0xa8, 0x0, 0xa}, Mask:net.IPMask{0xff, 0xff, 0xff, 0x0}}
2009/11/10 23:00:00 new: net.IPNet{IP:net.IP{0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0xff, 0xff, 0xc0, 0xa8, 0x0, 0xa}, Mask:net.IPMask{0xff, 0xff, 0xff, 0x0}}
```
We moved from IPv4 to IPv6.